modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/KotlinClientCodegen.java

```diff
@@ -1,27 +1,32 @@
 package io.swagger.codegen.languages;
 
-import io.swagger.codegen.CliOption;
-import io.swagger.codegen.CodegenConstants;
-import io.swagger.codegen.CodegenType;
-import io.swagger.codegen.SupportingFile;
+import io.swagger.codegen.*;
+import io.swagger.models.Model;
+import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.File;
-import java.util.HashMap;
-import java.util.Map;
+import java.text.Collator;
+import java.util.*;
+
+import static io.swagger.codegen.languages.JavaClientCodegen.prioritizeContentTypes;
+import static java.util.Collections.sort;
 
 public class KotlinClientCodegen extends AbstractKotlinCodegen {
 
     public static final String DATE_LIBRARY = "dateLibrary";
-    protected CodegenConstants.ENUM_PROPERTY_NAMING_TYPE enumPropertyNaming = CodegenConstants.ENUM_PROPERTY_NAMING_TYPE.camelCase;
+    public static final String USE_ASYNC = "useAsync";
+    private static final String RETROFIT_2 = "retrofit2";
+
     static Logger LOGGER = LoggerFactory.getLogger(KotlinClientCodegen.class);
 
-    protected String dateLibrary = DateLibrary.JAVA8.value;
+    protected String dateLibrary = DateLibrary.JAVA8_LOCAL.value;
 
     public enum DateLibrary {
         STRING("string"),
         THREETENBP("threetenbp"),
+        JAVA8_LOCAL("java8-localdatetime"),
```

Comment:
I don't feel like this is an appropriate use of `DateLibrary`. The library is Java 8, you're just encoding whether you want local dates vs OffsetDateTime. This is type of thing is usually handled via CLI options or `additional-properties`. Was there another generator encoding options into the library enum in this way?

Reply:
Changed: added as  CLI option